### PR TITLE
[MI-2750]: Implemented fetching channel list on webapp and removes from plugin's server.

### PR DESCRIPTION
**Pull request by avas27JTG**
Implemented fetching channel list on webapp and removes from plugin's server.

### CODE REVIEW

File: webapp/src/hooks/useMattermostApi.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import {useSelector, useDispatch} from 'react-redux';
+
+import {mattermostServerApi} from 'services';
+
+function useMattermostApi() {
+    const state = useSelector((reduxState: ReduxState) => reduxState);
+    const dispatch = useDispatch();
+
+    // Pass payload only in POST rquests for GET requests there is no need to pass payload argument
```
Comment (ayusht2810) [inline webapp/src/hooks/useMattermostApi.ts:9]:
```suggestion
    // Pass payload only in POST requests, for GET requests there is no need to pass payload argument
```

File: webapp/src/hooks/useMattermostApi.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    const makeMattermostApiRequest = async (serviceName: MattermostApiServiceName, payload: FetchChannelParams): Promise<any> => {
+        return dispatch(mattermostServerApi.endpoints[serviceName].initiate(payload)); //TODO: add proper type here
+    };
+
+    // Pass payload only in POST rquests for GET requests there is no need to pass payload argument
```
Comment (ayusht2810) [inline webapp/src/hooks/useMattermostApi.ts:14]: same

File: webapp/src/services/serverApi.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import {createApi, fetchBaseQuery} from '@reduxjs/toolkit/query/react';
+
+import Cookies from 'js-cookie';
+
+import Constants from 'pluginConstants';
+import Utils from 'utils';
+
+// Service to make plugin API requests
```
Comment (ayusht2810) [inline webapp/src/services/serverApi.ts:8]:
```suggestion
// Service to make mattermost API requests
```

File: webapp/src/services/pluginApi.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+// Service to make plugin API requests
+export const azureDevOpsPluginApi = createApi({
+    reducerPath: 'azureDevOpsPluginApi',
+    baseQuery: fetchBaseQuery({baseUrl: Utils.getBaseUrls().pluginApiBaseUrl}),
+    tagTypes: ['Posts'],
+    endpoints: (builder) => ({
+        [Constants.pluginApiServiceConfigs.createTask.apiServiceName]: builder.query<void, APIRequestPayload>({
+            query: (payload) => ({
+                headers: {[Constants.common.HeaderCSRFToken]: Cookies.get(Constants.common.MMCSRF)},
```
Comment (ayusht2810) [inline webapp/src/services/pluginApi.ts:16]: should we use the headers at `prepareHeaders` as used in `serverApi.ts` file?

File: webapp/src/services/serverApi.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+            return headers;
+        },
+    }),
+    tagTypes: ['Posts'],
+    endpoints: (builder) => ({
+        [Constants.mattermostApiServiceConfigs.getChannels.apiServiceName]: builder.query<ChannelList[], FetchChannelParams>({
+            query: (params) => {
+                const currentUserId = Cookies.get(Constants.common.MMUSERID) ?? '';
+                return ({
+                    headers: {[Constants.common.HeaderCSRFToken]: Cookies.get(Constants.common.MMCSRF)},
```
Comment (ayusht2810) [inline webapp/src/services/serverApi.ts:29]: same

**Review by ayusht2810 — APPROVED** [review]

**Review by manojmalik20 — APPROVED** [review]

**Pull request merged**